api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -836,6 +859,90 @@ func (s *SolutionManager) Reconcil() []error {
 	return nil
 }
 
+func (s *SolutionManager) InitCancelMap() {
```

Comment:
You don't need to expose a separate init function for cancel map, please embed it in solution-manager constructor (init).